onnxruntime/contrib_ops/cpu/transformers/beam_search_impl_gpt.h

```diff
@@ -347,15 +355,20 @@ Status BeamSearchGpt<T>::Execute(const FeedsFetchesManager* init_run_feeds_fetch
       ORT_RETURN_IF_ERROR(UpdateFeeds(fetches, feeds, current_length,
                                       position_ids, increase_position,
                                       ReinterpretAsSpan<const int32_t>(beam_next_tokens),
-                                      ReinterpretAsSpan<const int32_t>(beam_indices),
                                       gpt_subgraph_.has_decoder_masked_attention_
-                                          ? ReinterpretAsSpan<const int32_t>(beam_state.chosen_indices)
+                                          ? place_holder
+                                          : ReinterpretAsSpan<const int32_t>(this->beam_scorer_->GetNextIndicesCPU()),
+                                      gpt_subgraph_.has_decoder_masked_attention_
+                                          ? ReinterpretAsSpan<const int32_t>(this->beam_scorer_->GetNextIndicesGPU())
                                           : place_holder,
                                       current_length - 1,
                                       parameters->sequence_length,
                                       gpt_subgraph_.has_decoder_masked_attention_));
     }
 
+    if (this->beam_scorer_->IsDoneLater())
```

Comment:
Is there some performance benefit to separate IsDoneLater from IsDone? Compared to IsDone, it will do an extra UpdateFeeds, but might reduce some wait time in synchronize.